bundles/org.openhab.binding.somfymylink/src/main/java/org/openhab/binding/somfymylink/internal/discovery/SomfyMyLinkDeviceDiscoveryService.java

```diff
@@ -0,0 +1,196 @@
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkScene;
+import org.openhab.binding.somfymylink.internal.model.SomfyMyLinkShade;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Deactivate;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link SomfyMyLinkDeviceDiscoveryService} is responsible discovering things connected to the mylink.
+ *
+ * @author Chris Johnson - Initial contribution
+ */
+@NonNullByDefault
+public class SomfyMyLinkDeviceDiscoveryService extends AbstractDiscoveryService {
+
+    private static final int DISCOVERY_REFRESH_SEC = 1800;
+
+    private final Logger logger = LoggerFactory.getLogger(SomfyMyLinkDeviceDiscoveryService.class);
+
+    private @Nullable ScheduledFuture<?> scanTask;
+
+    private SomfyMyLinkBridgeHandler mylinkHandler;
+
+    private @Nullable ScheduledFuture<?> discoveryJob;
+
+    public SomfyMyLinkDeviceDiscoveryService(SomfyMyLinkBridgeHandler mylinkHandler) throws IllegalArgumentException {
+        super(SomfyMyLinkHandlerFactory.DISCOVERABLE_DEVICE_TYPES_UIDS, 10);
+
+        this.mylinkHandler = mylinkHandler;
+    }
+
+    @Override
+    @Activate
+    public void activate(@Nullable Map<@NonNull String, @Nullable Object> configProperties) {
+        super.activate(configProperties);
+    }
+
+    @Override
+    @Deactivate
+    public void deactivate() {
+        super.deactivate();
+    }
+
+    @Override
+    protected void startBackgroundDiscovery() {
+        logger.debug("Starting Somfy My Link background discovery");
+
+        if (discoveryJob == null || discoveryJob.isCancelled()) {
+            discoveryJob = scheduler.scheduleWithFixedDelay(this::runDiscovery, 10, DISCOVERY_REFRESH_SEC,
+                    TimeUnit.SECONDS);
+        }
+    }
+
+    @Override
+    protected void stopBackgroundDiscovery() {
+        logger.debug("Stopping Somfy MyLink background discovery");
+        if (discoveryJob != null && !discoveryJob.isCancelled()) {
+            discoveryJob.cancel(true);
+            discoveryJob = null;
+        }
+    }
+
+    @Override
+    protected void startScan() {
+        runDiscovery();
+    }
+
+    private synchronized void runDiscovery() {
+        logger.debug("Starting scanning for things...");
+
+        if (this.scanTask == null || this.scanTask.isDone()) {
+            this.scanTask = scheduler.schedule(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        discoverDevices();
+                    } catch (SomfyMyLinkException e) {
+                        logger.info("Error scanning for devices: {}", e.getMessage(), e);
+
+                        if (scanListener != null) {
+                            scanListener.onErrorOccurred(e);
+                        }
+                    }
+                }
+            }, 0, TimeUnit.SECONDS);
```

Comment:
```suggestion
            this.scanTask = scheduler.execute(() -> {
                    try {
                        discoverDevices();
                    } catch (SomfyMyLinkException e) {
                        logger.info("Error scanning for devices: {}", e.getMessage(), e);

                        if (scanListener != null) {
                            scanListener.onErrorOccurred(e);
                        }
                    }
            });
```

Reply:
i removed scanTask all together as execute() returns void. i'm not concerned about multiple of these running as the comms with the bridge has a timeout and so the task will fail.